core/src/main/java/io/opencensus/stats/StatsComponent.java

```diff
@@ -13,21 +13,57 @@
 
 package io.opencensus.stats;
 
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
 /**
  * Class that holds the implementations for {@link ViewManager}, {@link StatsRecorder}, and {@link
  * StatsContextFactory}.
  *
  * <p>All objects returned by methods on {@code StatsComponent} are cacheable.
  */
-// TODO(sebright): Add a no-op StatsComponent.
 public abstract class StatsComponent {
+  private static final StatsComponent NOOP_STATS_COMPONENT = new NoopStatsComponent();
+
   /** Returns the default {@link ViewManager}. */
+  @Nullable
   public abstract ViewManager getViewManager();
 
   /** Returns the default {@link StatsRecorder}. */
   public abstract StatsRecorder getStatsRecorder();
 
   /** Returns the default {@link StatsContextFactory}. */
   // TODO(sebright): Remove this method once StatsContext is replaced by TagContext.
+  @Nullable
   abstract StatsContextFactory getStatsContextFactory();
+
+  /**
+   * Returns a {@code StatsComponent} that has a no-op implementation for the {@link StatsRecorder}.
+   *
+   * @return a {@code StatsComponent} that has a no-op implementation for the {@code StatsRecorder}.
+   */
+  static StatsComponent getNoopStatsComponent() {
+    return NOOP_STATS_COMPONENT;
+  }
+
+  @Immutable
+  private static final class NoopStatsComponent extends StatsComponent {
+
+    @Override
+    @Nullable
+    public ViewManager getViewManager() {
+      return null;
```

Comment:
I'm not sure if we should create a no-op ViewManager, because it could give the user incorrect results, but I added a TODO for us to decide whether to add one.